src/autocomplete/zsh.ts

```diff
@@ -133,7 +133,7 @@ _${this.config.bin}
       // skip hidden flags
       if (f.hidden) continue
 
-      f.summary = sanitizeSummary(f.summary || f.description)
+      const flagSummary = sanitizeSummary(f.summary || f.description)
```

Comment:
This is done to avoid mutating the `flag` object from `this.config`.
`sf autocomplete` will create completion files for all shells, even if it detects which one you are using:
https://github.com/oclif/plugin-autocomplete/blob/cd/powershell/src/commands/autocomplete/create.ts#L66-L68

The issue is that we are passing around `this.config` instead of a deep clone, so any modifications to it in the zsh class will affect the powershell one too.

I found it hard to get a clone of `this.config` with the common suggestions b/c of getters/deep classes, left as todo (WI).